File: web/src/api/jxc/goods.js
import request from '@/utils/request'

// ========== 商品 SPU ==========
export const getGoodsPage = (params) =>
  request({ url: '/jxc/goods/page', method: 'get', params })

export const getGoodsDetail = (params) =>
  request({ url: '/jxc/goods/detail', method: 'get', params })

export const getAllGoods = () =>
  request({ url: '/jxc/goods/all', method: 'get' })

export const createGoods = (data) =>
  request({ url: '/jxc/goods', method: 'post', data })

export const updateGoods = (data) =>
  request({ url: '/jxc/goods', method: 'put', data })

export const deleteGoods = (data) =>
  request({ url: '/jxc/goods', method: 'delete', data })

export const setGoodsStatus = (data) =>
  request({ url: '/jxc/goods/status', method: 'put', data })

// ========== 商品 SKU ==========
export const getSkuList = (params) =>
  request({ url: '/jxc/goods/sku/list', method: 'get', params })

export const createSku = (data) =>
  request({ url: '/jxc/goods/sku', method: 'post', data })

export const updateSku = (data) =>
  request({ url: '/jxc/goods/sku', method: 'put', data })

export const deleteSku = (data) =>
  request({ url: '/jxc/goods/sku', method: 'delete', data })

export const setSkuStatus = (data) =>
  request({ url: '/jxc/goods/sku/status', method: 'put', data })

File: server/router/jxc/goods.go
package jxc

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type GoodsRouter struct{}

func (r *GoodsRouter) InitGoodsRouter(Router *gin.RouterGroup) {
	recordGroup := Router.Group("jxc").Use(middleware.OperationRecord())
	readOnlyGroup := Router.Group("jxc")

	// ===== 商品 SPU =====
	readOnlyGroup.GET("goods/page", goodsApi.GetGoodsPage)
	readOnlyGroup.GET("goods/detail", goodsApi.GetGoodsDetail)
	readOnlyGroup.GET("goods/all", goodsApi.GetAllGoods)
	recordGroup.POST("goods", goodsApi.CreateGoods)
	recordGroup.PUT("goods", goodsApi.UpdateGoods)
	recordGroup.DELETE("goods", goodsApi.DeleteGoods)
	recordGroup.DELETE("goods/forever", goodsApi.DeleteGoodsForever)
	recordGroup.PUT("goods/status", goodsApi.SetGoodsStatus)

	// ===== 商品 SKU =====
	readOnlyGroup.GET("goods/sku/list", goodsApi.GetSkuList)
	readOnlyGroup.GET("goods/sku/by-barcode", goodsApi.GetSkuByBarcode)
	recordGroup.POST("goods/sku", goodsApi.CreateSku)
	recordGroup.PUT("goods/sku", goodsApi.UpdateSku)
	recordGroup.DELETE("goods/sku", goodsApi.DeleteSku)
	recordGroup.DELETE("goods/sku/forever", goodsApi.DeleteSkuForever)
	recordGroup.PUT("goods/sku/status", goodsApi.SetSkuStatus)
}
